pkg/controllers/job/job_controller_actions.go

```diff
@@ -452,6 +468,78 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 	return nil
 }
 
+func (cc *jobcontroller) waitDependsOnTaskMeetCondition(taskName string, podToCreateEachTask []*v1.Pod, job *batch.Job) {
+	taskIndex := jobhelpers.GetTaskIndex(taskName, job)
+	if job.Spec.Tasks[taskIndex].DependsOn != nil {
+		dependsOn := *job.Spec.Tasks[taskIndex].DependsOn
+		if len(dependsOn.Name) > 1 && dependsOn.Iteration == batch.IterationAny {
+			wait.PollInfinite(100*time.Millisecond, func() (bool, error) {
+				for _, DependsOnTask := range dependsOn.Name {
+					if cc.isDenpendentPodsReady(DependsOnTask, job) {
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+		} else {
+			for _, DependsOnTask := range dependsOn.Name {
+				wait.PollInfinite(100*time.Millisecond, func() (bool, error) {
+					if cc.isDenpendentPodsReady(DependsOnTask, job) {
+						return true, nil
+					}
+					return false, nil
+				})
+			}
+		}
+	}
+}
+
+func (cc *jobcontroller) isDenpendentPodsReady(DependsOnTask string, job *batch.Job) bool {
+	dependentPod := jobhelpers.GetPodsInTask(DependsOnTask, job)
+	dependsOnTaskIndex := jobhelpers.GetTaskIndex(DependsOnTask, job)
+	runningPodCount := 0
+	for _, podName := range dependentPod {
+		pod, err := cc.podLister.Pods(job.Namespace).Get(podName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				_, err := cc.vcClient.BatchV1alpha1().Jobs(job.Namespace).Get(context.TODO(), job.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						return true
+					}
+				}
+				return false
+			}
+			klog.Errorf("Failed to get pod %v/%v %v", job.Namespace, podName, err)
+			return false
+		}
+
+		if pod.Status.Phase != v1.PodRunning {
+			klog.V(4).Infof("Sequential state, pod %v/%v of dependent tasks is not running", pod.Namespace, pod.Name)
+			continue
```

Comment:
what if all the pods under that task become succeeded quickly?

Reply:
The current situation is that the following tasks will stop and will never be run.
This should be dependent on completion status, we can add a field later:

```
dependsOn:
    name: xxxxx
    status: running/complete
```


I think we can also deal with it now, if it is completed, allow it to pass.

```go
pod.Status.Phase != v1.PodRunning || pod.Status.Phase != v1.PodSucceeded
```